sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -188,6 +190,30 @@ class CSVSuite extends QueryTest with SharedSparkSession with TestCsvData {
     verifyCars(cars, withHeader = true)
   }
 
+  test("test with tab delimiter and double quote") {
+    val cars = spark.read
+        .options(Map("quote" -> "\"", "delimiter" -> """\t""", "header" -> "true"))
+        .csv(testFile(carsTsvFile))
+
+    verifyCars(cars, numFields = 6, withHeader = true, checkHeader = false)
+  }
+
+  test("SPARK-24540: test with multiple character delimiter (comma space)") {
+    val cars = spark.read
+        .options(Map("quote" -> "\'", "delimiter" -> ", ", "header" -> "true"))
+        .csv(testFile(carsMultiCharDelimitedFile))
+
+    verifyCars(cars, withHeader = true)
+  }
+
+  test("SPARK-24540: test with multiple (crazy) character delimiter") {
+    val cars = spark.read
+        .options(Map("quote" -> "\'", "delimiter" -> """_/-\\_""", "header" -> "true"))
```

Comment:
Hm, wait, one more possible issue. This specifies the delimiter as string `_/-\_`, which should then be unescaped to `_/-_`, which isn't what your test case expects. The test passes because I don't think it checks anything but year? Hm, I actually trying to figure out how it works at all in this case, as the CSV file uses `_/-\_` as the delimiter. @jeff303 am I missing something or is there a test issue?

Reply:
I added some additional assertions for the rest of the columns (besides year).  The delimiter is given as `_/-\\_`.  This is needed because the fourth character of this delimiter is a literal backslash.  So after unescaping (via the new utility method), we end up with `_/-\_` as the String that is passed into the Univocity parser.

Reply:
Oh right, Scala doesn't unescape it because of `"""`. This makes sense then. Good to have more tests.